packages/roosterjs-editor-core/lib/undo/Undo.ts

```diff
@@ -147,24 +146,31 @@ export default class Undo implements UndoService {
         // Handle backspace/delete when there is a selection to take a snapshot
         // since we want the state prior to deletion restorable
         let evt = pluginEvent.rawEvent as KeyboardEvent;
-        let shouldTakeUndo = false;
         if (evt.which == KEY_BACKSPACE || evt.which == KEY_DELETE) {
             let selectionRange = this.editor.getSelectionRange();
+
+            // Add snapshot when
+            // 1. Something has been selected (not collapsed), or
+            // 2. This is not the same operation with previous one, or
```

Comment:
What does 'This is not the same operation with previous one' mean? Also why do we leave out image?

Reply:
I updated the comment. This is to avoid adding to many snapshot when user keep pressing delete/backspace, we only add a snapshot before the first press.

For image, it can already be covered by "this.lastKeyPress != evt.which", so we don't need to check it separately.